Sources/Filters/General/ImageDataOutlineFilter/index.js

```diff
@@ -0,0 +1,143 @@
+import macro from 'vtk.js/Sources/macros';
+import vtkBoundingBox from 'vtk.js/Sources/Common/DataModel/BoundingBox';
+import vtkCellArray from 'vtk.js/Sources/Common/Core/CellArray';
+import vtkCubeSource from 'vtk.js/Sources/Filters/Sources/CubeSource';
+import vtkPolyData from 'vtk.js/Sources/Common/DataModel/PolyData';
+import vtkTransform from 'vtk.js/Sources/Common/Transform/Transform';
+
+const { vtkErrorMacro } = macro;
+
+// prettier-ignore
+export const LINE_ARRAY = [
+  2, 0, 1,
+  2, 2, 3,
+  2, 4, 5,
+  2, 6, 7,
+  2, 0, 2,
+  2, 1, 3,
+  2, 4, 6,
+  2, 5, 7,
+  2, 0, 4,
+  2, 1, 5,
+  2, 2, 6,
+  2, 3, 7,
+];
+
+// ----------------------------------------------------------------------------
+// vtkImageDataOutlineFilter methods
+// ----------------------------------------------------------------------------
+
+function vtkImageDataOutlineFilter(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkImageDataOutlineFilter');
+
+  publicAPI.requestData = (inData, outData) => {
+    // implement requestData
+    const input = inData[0];
+
+    if (!input || !input.isA('vtkImageData')) {
+      vtkErrorMacro('Invalid or missing input');
+      return;
+    }
+
+    if (!model._tmpOut) {
+      // allocate output object if not done previously.
+      model._tmpOut = vtkPolyData.newInstance();
+    }
+
+    // First create a cube polydata in the index-space of the image.
+    const spatialExt = input.getSpatialExtent();
+    if (!spatialExt) {
+      vtkErrorMacro('Unable to fetch spatial extents of input image.');
+      return;
+    }
+
+    model._boundingBox.setBounds(spatialExt);
+    const lengths = model._boundingBox.getLengths();
+    model._cubeSource.setXLength(lengths[0]);
+    model._cubeSource.setYLength(lengths[1]);
+    model._cubeSource.setZLength(lengths[2]);
+    model._cubeSource.setCenter(model._boundingBox.getCenter());
+    model._cubeSource.update();
+
+    const out = model._cubeSource.getOutputData();
+    model._tmpOut.getPoints().deepCopy(out.getPoints());
+
+    // Now, transform the cube polydata points in-place
+    // using the image's indexToWorld transformation.
+    model._transform.setMatrix(input.getIndexToWorld());
```

Comment:
vtkCubeSource already supports bounds "rotations".
Problem is, it supports only X, Y and Z angles.
It would be a good opportunity to extend the rotation to also support a matrix (to apply before the rotations ?)
You wouldn't have to worry here about transforming the points manually. Code would be more factorized.

Reply:
I feel like the matrix should be applied the last, as something similar to `setUserMatrix()` function from VTK. Also, the x,y,z rotations wouldn't make sense after the cube has been moved away from the origin.
So basically, the `rotations` and `center` values will provide the initial setup in _model coordinates_, and then the `setMatrix()` option can provide additional option for _world coordinates_ placement.